spec/webvalve/service_url_converter_spec.rb

```diff
@@ -0,0 +1,141 @@
+      end
+
+      it "doesn't match when missing the trailing slash tho" do
+        expect("http://bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a totally wildcarded protocol" do
+      let(:url) { "*://bar.com" }
+
+      it "matches http" do
+        expect("http://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches anything else" do
+        expect("gopher://bar.com/").to match(subject.regexp)
+      end
+    end
+
+    context "with a wildcarded partial protocol" do
+      let(:url) { "http*://bar.com" }
+
+      it "matches empty" do
+        expect("http://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches full" do
+        expect("https://bar.com/").to match(subject.regexp)
+      end
+    end
+
+    context "with a TLD that is a substring of another TLD" do
+      let(:url) { "http://bar.co" }
+
+      it "doesn't match a different TLD when extending" do
+        expect("http://bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a wildcard subdomain" do
+      let(:url) { "http://*.bar.com" }
+
+      it "matches" do
+        expect("http://foo.bar.com").to match(subject.regexp)
+      end
+
+      it "doesn't match when too many subdomains" do
+        expect("http://beep.foo.bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a partial postfix wildcard subdomain" do
+      let(:url) { "http://foo*.bar.com" }
+
+      it "matches when present" do
+        expect("http://foobaz.bar.com").to match(subject.regexp)
+      end
+
+      it "matches when empty" do
+        expect("http://foo.bar.com").to match(subject.regexp)
+      end
+
+      it "doesn't match when out of order" do
+        expect("http://bazfoo.bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a partial prefix wildcard subdomain" do
+      let(:url) { "http://*baz.bar.com" }
+
+      it "matches when present" do
+        expect("http://foobaz.bar.com").to match(subject.regexp)
+      end
+
+      it "doesn't match when empty" do
+        expect("http://.bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a wildcarded basic auth url" do
+      let(:url) { "http://*:*@bar.com" }
+
+      it "matches when present" do
+        expect("http://bilbo:baggins@bar.com").to match(subject.regexp)
+      end
+
+      it "doesn't match when malformed" do
+        expect("http://bilbobaggins@bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a wildcarded path" do
+      let(:url) { "http://bar.com/*/whatever" }
+
+      it "matches with arbitrarily spicy but legal, non-URL-significant characters" do
+        expect("http://bar.com/a0-_~[]!$'(),;%+/whatever").to match(subject.regexp)
+      end
+
+      it "doesn't match when you throw a URL-significant char in there" do
+        expect("http://bar.com/life=love/whatever").not_to match(subject.regexp)
```

Comment:
And conversely if you try to sneak in a boundary char. Interestingly this is another valid URL that would have to be URL Escaped at the app layer in order to wildcard it successfully.